ffmpeg/ffmpeg_test.go

```diff
@@ -1927,3 +1927,248 @@ PTS_EOF
   `
 	run(cmd)
 }
+
+func TestTranscoder_Rotation(t *testing.T) {
+	runRotationTests(t, Software)
+	// TODO hevc
+}
+
+func runRotationTests(t *testing.T, accel Acceleration) {
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+
+	// generate a sample that is rotated mid-stream
+	cmd := `
+		ffmpeg -i "$1/../transcoder/test.ts" -an -c:v libx264 -g 120 -s 100x56 -f segment -t 6 test-%d.ts
+		ffmpeg -i test-1.ts -vf transpose -c:v libx264 -c:a copy -copyts -muxdelay 0 test-1-transposed.ts
+		ffprobe -select_streams v -show_entries format=start_time,duration:stream=width,height -of default=nw=1 test-1.ts > test-1.data
+		ffprobe -select_streams v -count_frames -show_entries format=start_time,duration:stream=width,height,nb_read_frames -of default=nw=1 test-1-transposed.ts > test-1-transposed.data
+
+		cat <<-EOF1 > test-1.expected
+			width=100
+			height=56
+			width=100
+			height=56
+			start_time=3.433333
+			duration=2.000000
+		EOF1
+
+		# transposed
+		cat <<-EOF2 > test-1-transposed.expected
+			width=56
+			height=100
+			nb_read_frames=120
+			width=56
+			height=100
+			nb_read_frames=120
+			start_time=3.433333
+			duration=2.000000
+		EOF2
+
+		diff -u test-1.expected test-1.data
+		diff -u test-1-transposed.expected test-1-transposed.data
+
+		cat test-0.ts test-1-transposed.ts test-2.ts > double-rotated.ts
+		cat test-0.ts test-1-transposed.ts > single-rotated.ts
+	`
+	run(cmd)
+
+	profile := P144p30fps16x9
+	profilePassthrough := profile
+	profilePassthrough.Framerate = 0
+	res, err := Transcode3(
+		&TranscodeOptionsIn{Fname: dir + "/double-rotated.ts", Accel: accel},
+		[]TranscodeOptions{{
+			Profile: profile,
+			Oname:   dir + "/out-double-rotated-30fps.ts",
+			Accel:   accel,
+		}, {
+			Profile: profilePassthrough,
+			Oname:   dir + "/out-double-rotated.ts",
+			Accel:   accel,
+		}})
+	require.NoError(t, err)
+
+	assert.Equal(t, 360, res.Decoded.Frames)
+	assert.Equal(t, 181, res.Encoded[0].Frames) // should be 180 ... ts rounding ?
+	assert.Equal(t, 360, res.Encoded[1].Frames)
+
+	// TODO test rollover of gop interval during flush
```

Comment:
I'm not sure what we normally do in this repo for TODOs but you maybe want to raise tickets instead if they're important to come back to so we don't forget.

Reply:
this specific note is not so important, it is more noting a gap in the test coverage

the other TODOs sprinkled around these tests relate to discrepancies between cpu/gpu behavior are likely related to the green screen flash which we already have ticket(s) for